viamonvif/discovery.go

```diff
@@ -163,6 +200,26 @@ func DiscoverCameraInfo(
 	return zero, fmt.Errorf("no credentials matched IP %s", xaddr)
 }
 
+var consecutiveDashesRegexp = regexp.MustCompile(`\-\-+`)
+
+func strToHostName(inp string) string {
```

Comment:
could you add a comment with a link to the spec / standard this is comfirming to?

Reply:
Are you asking about the spec that defines what makes a valid DNS name?

Or [this line](https://docs.google.com/document/d/1bxObh_p7Ix94gEhmvwnPDW23TzJp21NptaadMRqXG5A/edit?tab=t.0#bookmark=id.rrxrkcoj02ig) that describes we want the property where the same input will continue to map to the same output?

The intention of the immediately following documentation was to copy the necessary context over. I'm happy to add a blurb about predictability and more explicitly state this is all of the context behind this algorithmic decision.

Reply:
valid DNS name.